test/semantics/dosemantics11.f90

```diff
@@ -0,0 +1,341 @@
+! Copyright (c) 2019, NVIDIA CORPORATION.  All rights reserved.
+!
+! Licensed under the Apache License, Version 2.0 (the "License");
+! you may not use this file except in compliance with the License.
+! You may obtain a copy of the License at
+!
+!     http://www.apache.org/licenses/LICENSE-2.0
+!
+! Unless required by applicable law or agreed to in writing, software
+! distributed under the License is distributed on an "AS IS" BASIS,
+! WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+! See the License for the specific language governing permissions and
+! limitations under the License.
+!
+! C1135 A cycle-stmt shall not appear within a CHANGE TEAM, CRITICAL, or DO 
+! CONCURRENT construct if it belongs to an outer construct.
+!
+! C1167 -- An exit-stmt shall not appear within a DO CONCURRENT construct if 
+! it belongs to that construct or an outer construct.
+!
+! C1168 -- An exit-stmt shall not appear within a CHANGE TEAM or CRITICAL 
+! construct if it belongs to an outer construct.
+
+subroutine s1()
+!ERROR: No matching construct for CYCLE statement
+  cycle
+end subroutine s1
+
+subroutine s2()
+!ERROR: No matching construct for EXIT statement
+  exit
+end subroutine s2
+
+subroutine s3()
+  level0: block
+!ERROR: No matching construct for CYCLE statement
+    cycle level0
```

Comment:
It's confusing to say there is not matching construct because there is a block construct with this name. I think it would be better to say that the CYCLE statement must name a DO construct.

Reply:
Good point.  I was trying to combine the error reporting for CYCLE and EXIT statements, but it's clearly better to differentiate them.  I'll do that.